src/ethereum_test_specs/helpers.py

```diff
@@ -132,21 +152,59 @@ def verify_transaction_exception(
             )
 
 
+def verify_transaction_receipt(
+    transaction_index: int,
+    expected_receipt: TransactionReceipt | None,
+    actual_receipt: TransactionReceipt | None,
+):
+    """
+    Verify the actual receipt against the expected one.
+
+    If the expected receipt is None, validation is skipped.
+
+    Only verifies non-None values in the expected receipt if any.
+    """
+    if expected_receipt is None:
+        return
+    assert actual_receipt is not None
+    if (
+        expected_receipt.gas_used is not None
+        and actual_receipt.gas_used != expected_receipt.gas_used
+    ):
+        raise TransactionReceiptMismatchError(
+            index=transaction_index,
+            field_name="gas_used",
+            expected_value=expected_receipt.gas_used,
+            actual_value=actual_receipt.gas_used,
+        )
+    # TODO: Add more fields as needed
+
+
 def verify_transactions(
-    exception_mapper: ExceptionMapper, txs: List[Transaction], result: Result
+    *,
+    txs: List[Transaction],
+    exception_mapper: ExceptionMapper,
+    result: Result,
 ) -> List[int]:
     """
-    Verify rejected transactions (if any) against the expected outcome.
-    Raises exception on unexpected rejections or unexpected successful txs.
+    Verify accepted and rejected (if any) transactions against the expected outcome.
+    Raises exception on unexpected rejections, unexpected successful txs, or successful txs with
+    unexpected receipt values.
     """
     rejected_txs: Dict[int, str] = {
         rejected_tx.index: rejected_tx.error for rejected_tx in result.rejected_transactions
     }
 
+    receipt_index = 0
     for i, tx in enumerate(txs):
         error_message = rejected_txs[i] if i in rejected_txs else None
-        info = TransactionExceptionInfo(t8n_error_message=error_message, transaction_ind=i, tx=tx)
+        info = TransactionExceptionInfo(
+            t8n_error_message=error_message, transaction_index=i, tx=tx
+        )
         verify_transaction_exception(exception_mapper=exception_mapper, info=info)
```

Comment:
perhaps we should have unit test for transaction exception cases
occurred vs expected 
```
None        ==  None
None       == Exception
Exception  == None
Exception  == Exception
```

Reply:
Added unit test in latest commit.